source/jormungandr/jormungandr/instance.py

```diff
@@ -432,28 +432,37 @@ def max_nb_crowfly_by_mode(self):
         # the value by default is a dict...
         return copy.deepcopy(get_value_or_default('max_nb_crowfly_by_mode', instance_db, self.name))
 
+    def reap_socket(self, ttl):
+        if self.zmq_socket_type != 'transient':
+            return
+        logger = logging.getLogger(__name__)
+        now = time.time()
+        while True:
+            try:
+                socket, t = self._sockets.popleft()
+                if now - t > ttl:
+                    logger.debug("closing one socket for %s", self.name)
+                    socket.setsockopt(zmq.LINGER, 0)
+                    socket.close()
+                else:
+                    self._sockets.appendleft((socket, t))
```

Comment:
interesting trick... i suppose this line will help to distribute more evenly the load?

Reply:
it's not a trick, it's a feature :wink: 
In `socket()` we use the `dequeue` as a stack, so the most recently used socket will be on the rigth side of the `deque` while the least rencently used socket is on the left side.
Here we start by looking if we need to close the socket on the left side, if we encounter a socket that has been used too recently to be closed we can stop as all other sockets after this one has been used more recently that this one.
At this line, since the current socket has still some time left before beeing brutaly murdered, we put it back in place on the left side.